FIX: prevent raw._data overwrite from _add_exg and _add_chpi

jstout211 opened the pull request:
#### Reference issue
Fixes #8587 


#### What does this implement/fix?
in mne/simulation/raw.py  - functions _add_exg and _add_chpi, the code set the raw._data to a variable "data".
The variable "data" was written over by the output of these functions.  This would leave the raw._data as the last operation performed. The fix is to sum the raw._data numpy array with the output from the functions.  (+= vs =).

agramfort commented:
thx @jstout211 

can you add an entry in the what's new page?
also I wondering if we could add a test that would fail on master and that now works?

larsoner commented on mne/tests/test_simulation.py:
```diff
@@ -0,0 +1,63 @@
+from mne.datasets import sample
+
+
+def test_simulation_cascade():
```
Rather than creating a new file, could you put this in `mne/simulation/tests/test_raw.py`?

jstout211 replied:
Added test into the specified file and git rm-ed the mne/tests/test_simulation.py

larsoner commented on mne/tests/test_simulation.py:
```diff
@@ -0,0 +1,63 @@
+    serial_data = raw_eog._data + raw_ecg._data + raw_noise._data
+    # Should add chpi once fixed above
+
+    test = (cascade_data == serial_data)
+    assert test.all()
```
Exact equality with float is often iffy and this will give a less informative error than:
```
assert_allclose(cascade_data, serial_data)
```

larsoner reviewed (commented):
Looks like a good start, can you add an entry to `doc/latest.inc` in the BUG section? It looks like you are a new contributor so please try to follow the format of other new contributors and put your entry at the top of the section -- thanks!

jstout211 commented:
@larsoner - I tried to add a test for the add_chpi - but the assert_close fails.  I didn't see the ability to set a seed number on that simulation.  I backed out of that modification, so it currently tests eyeblink, cardiac, and noise addition.  add_chpi does not have testing coverage even though the code was changed in the same manner within simulation/raw.py.

larsoner commented:
> I didn't see the ability to set a seed number on that simulation.

There is no randomness in the cHPI so nothing to seed. Can you add the code that you think should pass but does not and I'll see if I can push a commit to fix it?

jstout211 commented:
> > I didn't see the ability to set a seed number on that simulation.
> 
> There is no randomness in the cHPI so nothing to seed. Can you add the code that you think should pass but does not and I'll see if I can push a commit to fix it?

@larsoner - I changed the test to use the chpi test dataset and added the chpi components to both signal routes.  It is not currently passing. 

![image](https://user-images.githubusercontent.com/2371720/102277589-81790100-3ef6-11eb-9408-5a01e085bd78.png)

larsoner commented on doc/changes/names.inc:
```diff
@@ -342,4 +342,6 @@
 
 .. _Hongjiang Ye: https://github.com/rubyyhj
 
+.. _Jeff Stout: https://megcore.nih.gov/index.php/Staff
```
@jstout211 let me know if you want a different URL (or if this is not you!), I chose this from 1 minute of Google-fu

larsoner replied:
@jstout211 I'm going to merge this PR but feel free to comment here if the name or URL should be changed

larsoner commented:
Your code more or less worked, once I changed to use `raw.copy()` and `raw.get_data()` rather than `deepcopy(raw)` and `raw._data` directly things worked. Not 100% sure why, maybe some weird numerical precision thing... things pass locally, let's see if CIs agree and then merge once you confirm or send a new URL above -- thanks in advance @jstout211
